omas/omas_utils.py

```diff
@@ -7,6 +7,213 @@
+        By convention, only the upper_error node should be filled in case of symmetrical error bars.
+        The upper and lower errors are absolute and defined positive, the effective values of the data will be within the interval [data-data_error_lower, data+data_error_upper].
+        Thus whatever the sign of data, data_error_lower relates to the lower bound and data_error_upper to the upper bound of the error bar interval.
+
+        :param nominal_value: nominal value of the uncertain quantity
+
+        :param error_upper: standard_deviation of symmetrical uncertainty, or upper error bound for asymmetrical one
+
+        :param error_lower: lower error bound for asymmetrical uncertainty
+
+        :param error_index: type of error bar description which is used for a given location
+
+        :param location: location from which this data was loaded
+        '''
+        self._nominal_value = nominal_value
+        assert error_upper is None or not isinstance(error_upper, str), 'error_upper must be a number'
+        self._error_upper = error_upper
+        assert error_lower is None or not isinstance(error_lower, str), 'error_lower must be a number'
+        self._error_lower = error_lower
+        assert error_index is None or isinstance(error_index, int), 'error_index must be a integer'
+        self._error_index = error_index
+        assert location is None or isinstance(location, str), 'location must be a string'
+        self._location = location
+
+    def __getstate__(self):
+        state = {}
+        for item in ['_nominal_value', '_error_upper', '_error_lower', '_error_index', '_location']:
+            state[item] = self.__dict__[item]
+        return state
+
+    def __setstate__(self, state):
+        for item in ['_nominal_value', '_error_upper', '_error_lower', '_error_index', '_location']:
+            self.__dict__[item] = state[item]
+        return self
+
+    @property
+    def location(self):
+        return self._location
+
+    @property
+    def nominal_value(self):
+        return self._nominal_value
+
+    @property
+    def error_upper(self):
+        if self._error_upper is not None:
+            return self._error_upper
+        else:
+            return self.nominal_value * 0.0
+
+    @property
+    def error_lower(self):
+        if self._error_lower is not None:
+            return self._error_lower
+        elif self._error_upper is not None:
+            return self._error_upper
+        else:
+            return self.nominal_value * 0.0
+
+    @property
+    def error_index(self):
+        return self._error_index
+
+    @property
+    def std_dev(self):
+        '''
+        Return standard deviation for symmetric uncertainty bounds
+        NOTE: This method will raise an error if uncertainty is not symmetrical
+        '''
+        if self._error_upper is None:
+            return self.nominal_value * 0.0
+        elif self._error_lower is None:
+            return self.error_upper
+        else:
```

Comment:
What about the case where `self.error_upper==self.error_lower`?

Reply:
Good point, but after some thought I think it is better to raise an error, the point being that if the error is symmetrical then only `error_upper` should be filled. It's better to raise these issues always, rather than having things pass in some cases but not others.

Reply:
As long as we thought about it, and made an active decision, I am OK.